libs/community/langchain_community/document_loaders/parsers/audio.py

```diff
@@ -12,14 +13,273 @@
+                Maximum number of retries for failed API requests.
+        """
+        self.api_key = api_key or os.environ.get("AZURE_OPENAI_API_KEY")
+        self.azure_endpoint = azure_endpoint or os.environ.get("AZURE_OPENAI_ENDPOINT")
+        self.api_version = api_version or os.environ.get("OPENAI_API_VERSION")
+        self.azure_ad_token = azure_ad_token
+
+        self.language = language
+        self.prompt = prompt
+        self.response_format = response_format
+        self.temperature = temperature
+
+        self.deployment_name = deployment_name
+        self.chunk_duration_threshold = chunk_duration_threshold
+        self.byte_limit = byte_limit
+        self.max_retries = max_retries
+        self.input_format = input_format
+
+    @property
+    def _create_params(self) -> Dict[str, Any]:
+        params = {
+            "language": self.language,
+            "prompt": self.prompt,
+            "response_format": self.response_format,
+            "temperature": self.temperature,
+        }
+        return {k: v for k, v in params.items() if v is not None}
+
+    def lazy_parse(self, blob: Blob) -> Iterator[Document]:
+        """
+        Lazily parse the provided audio blob for transcription.
+
+        Args:
+            blob (Blob):
+                The audio file in Blob format to be transcribed.
+
+        Yields:
+            Document:
+                Parsed transcription from the audio file.
+
+        Raises:
+            ImportError:
+                If the required packages `openai` or `pydub` are not installed.
+            ValueError:
+                If the file type is unsupported or exceeds the size limit.
+        """
+
+        try:
+            import openai
+        except ImportError:
+            raise ImportError(
+                "openai package not found, please install it with "
+                "`pip install openai`"
+            )
+
+        try:
+            from pydub import AudioSegment
+        except ImportError:
+            raise ImportError(
+                "pydub package not found, please install it with " "`pip install pydub`"
+            )
+
+        if not str(blob.path).endswith(self.input_format):
+            raise ValueError(
+                "File must be of one of the following types: "
+                f"{str(self.input_format)}"
+            )
+
+        if os.path.getsize(str(blob.source)) > self.byte_limit:
+            raise ValueError(
+                "Audio file exceeds 25 MB limit,"
+                f" length of file is {os.path.getsize(str(blob.source))} bytes"
+            )
+
+        if is_openai_v1():
+            client = openai.AzureOpenAI(
+                api_key=self.api_key,
+                azure_endpoint=self.azure_endpoint,
+                api_version=self.api_version,
+                max_retries=self.max_retries,
+                azure_ad_token=self.azure_ad_token,
+            )
+        else:
+            # Set the API key if provided
```

Comment:
if not using openai>=1, we also need to set api_type and api_version here:

openai.api_type = "azure"
openai.api_base = self.azure_endpoint
openai.api_key = self.api_key
openai.api_version = self.api_version

also, I don't believe openai<1 had a `base_url` property. This might be a bug in the openai whisper parser

Reply:
changed to 
```
        else:
            if self.api_key:
                openai.api_key = self.api_key
            if self.azure_endpoint:
                openai.api_base = self.azure_endpoint
            if self.api_version:
                openai.api_version = self.api_version
            openai.api_type = "azure"
```
I didn't realise this was wrong, would it make sense to suggest a change to the openai version then also?

Reply:
We can update the openai.base_url to openai.api_base for the openai parser.

Reply:
Sure thing! I'll make a note of that when I make the final PR too.